### DISCUSSION TITLE
Many-to-many relationship filtering

### DISCUSSION THREAD

**Discussion by MetaCorp** [supabase/supabase#26535] [General]
I want to create a note app with a tagging system. The notes are filterable by tags.

I have these tables in supabase:
1. note
2. tag
3. note_tag

I'm doing this query to filter by tags:
```javascript
let { data, error } = await supabase
      .from("note")
      .select(
        `
      *,
      note_tag!inner(*),
      tag!inner(*)
    `
      )
      .ilike("tag.text", `%${query}%`)
      .order("createdAt", {
        ascending: false,
      });
```

The problem is that this query does not return all the tags of a note, but only the matching one. I want to get the note that tags match the query but also get all the tags of that note in the response.

How to do so in supabase ?

**Comment by encima** [comment]
Hi @MetaCorp, I think the documentation [here](https://supabase.com/docs/guides/database/joins-and-nesting#specifying-the-on-clause-for-joins-with-multiple-foreign-keys) should help, does it?

2 other options are to:
1. Make 2 calls - one for the notes that have the tag and the second to retrieve all the note details
2. Write an RPC and call it from the JS SDK, something like:
```
CREATE OR REPLACE FUNCTION search_notes_with_tags(search_term TEXT)
RETURNS TABLE(note_id INT, note_content TEXT, tag_texts TEXT[]) AS $$
BEGIN
  RETURN QUERY
  SELECT
    n.id,
    n.content,
    ARRAY_AGG(t.text) AS tag_texts
  FROM
    notes n
    JOIN note_tags nt ON n.id = nt.note_id
    JOIN tags t ON t.id = nt.tag_id
  WHERE
    t.text ILIKE '%' || search_term || '%'
  GROUP BY
    n.id;
END;
$$ LANGUAGE plpgsql;
```

**Comment by MetaCorp** [comment]
Thank you for the answer. I managed to create an RPC to have the result intended. I used your query, but it still only returns the tag that matches and not all the tags of a note. Is it possible to do it in just one RPC call, or do I have to make two separates call for this ?

**Comment by Hallidayo** [comment]
Hi everyone, due to inactivity on this issue I've moved the issue over to discussions for future reference.

**Reply by GaryAustin1** [reply]
Are you meaning to put these in troubleshooting.  This is normally reserved for "official" Supabase troubleshooting that the AI is going to pick up.

**Reply by Hallidayo** [reply]
Ah right I didn't they would get picked up in that way. I'll change the category to General

**Comment by charlie-ttt** [comment]
@MetaCorp I'm having the same problem. Did you find a way to overcome this??
